public/app/features/provisioning/Repository/SyncRepository.tsx

```diff
@@ -13,31 +13,42 @@ interface Props {
 }
 
 export function SyncRepository({ repository }: Props) {
-  const [syncResource, syncQuery] = useCreateRepositorySyncMutation();
+  const [createJob, jobQuery] = useCreateRepositoryJobsMutation();
   const [isModalOpen, setIsModalOpen] = useState(false);
   const navigate = useNavigate();
   const name = repository.metadata?.name;
 
   useEffect(() => {
     const appEvents = getAppEvents();
-    if (syncQuery.isSuccess) {
+    if (jobQuery.isSuccess) {
       appEvents.publish({
         type: AppEvents.alertSuccess.name,
         payload: ['Pull started'],
       });
-    } else if (syncQuery.isError) {
+    } else if (jobQuery.isError) {
       appEvents.publish({
         type: AppEvents.alertError.name,
-        payload: ['Error pulling resources', syncQuery.error],
+        payload: ['Error pulling resources', jobQuery.error],
       });
     }
-  }, [syncQuery.error, syncQuery.isError, syncQuery.isSuccess]);
+  }, [jobQuery.error, jobQuery.isError, jobQuery.isSuccess]);
 
   const onClick = () => {
     if (!name) {
       return;
     }
-    syncResource({ name, body: { incremental: false } }); // will queue a full resync job
+    createJob({
+      name,
+      job: {
+        spec: {
+          action: 'pull',
+          repository: name,
```

Comment:
Do we need repository name in the spec, we already have it in the path?

Reply:
the openapi says it is required, so I put it in 😄   The value is populated by the one in the path

Let me see if I can make it optional/omitempty

Reply:
I was thinking more if you can omit it from the spect and it read it from the path on backend (not sure if that's possible).

Reply:
that *is* what happens -- but the spec counts it as a required property.  This is the the downside for having the same schema for read+write 🤷🏻 but we can make it work